src/buildApplication/generateFunctionsMap.ts

```diff
@@ -436,3 +433,25 @@ type LooseNode = Node & {
   */
 	value: any;
 };
+
+function getChunkIdentifier(chunkKey: number): string {
+	return `__chunk_${chunkKey}`;
+}
+
+function getChunkImportFn(functionPath: string): (chunkKey: number) => string {
+	const functionNestingLevel = getFunctionNestingLevel(functionPath);
+	const accountForNestingPath = `../`.repeat(functionNestingLevel);
+	return chunkKey => {
+		const chunkIdentifier = getChunkIdentifier(chunkKey);
+		const chunkPath = `${accountForNestingPath}__next-on-pages-dist__/chunks/${chunkKey}.js`;
+		return `import ${chunkIdentifier} from '${chunkPath}'`;
+	};
+}
+
+function getFunctionNestingLevel(functionPath: string): number {
+	const relativePath = functionPath.replace(
```

Comment:
https://nodejs.org/api/path.html#pathrelativefrom-to might help here.

Reply:
awesome! thanks! let me give that a try! 🤩👍

Reply:
I've applied the change 🙂  https://github.com/cloudflare/next-on-pages/pull/191/commits/6123deb2b8ca967c93af9e93f2f9b00f33dc1e44

it's a bit more verbose, but I think it's much more clean and robust than before 😄